subprojects/docs/src/docs/userguide/javaPlugin.adoc

```diff
@@ -563,6 +563,8 @@ The task has an `ignoreFailures` property to control the behavior when tests fai
 
 The `testLogging` property allows you to configure which test events are going to be logged and at which detail level. By default, a concise message will be logged for every failed test. See api:org.gradle.api.tasks.testing.logging.TestLoggingContainer[] for how to tune test logging to your preferences.
 
+Projects with large test suite can take a long time to execute even though a failure occurred early on leading to unnecessary wait times (especially on CI). To short circuit this behavior, the api:org.gradle.api.tasks.testing.Test#getFailFast()[`failFast`] property allows you to cause the `test` task to fail after the first test failure instead of running all tests. When this property is true, the resulting output will only show the results of tests that have completed up to and including the failure. The `--fail-fast` command line option enables the behavior from the command line.  Conversely, the '--no-fail-fast' command line option disables the behavior; this option is useful if the build file enables `failFast`, but you need to override it on a temporary basis. The default value for the `failFast` property is false.
```

Comment:
:x: This is outdated either way. `--no-fail-fast` doesn't exist anymore.

Reply:
Done.  I thought I had taken this out when I killed it in the design doc.  Thanks catching this.